src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+namespace SarifViewer
+{
+    public class ErrorListService
+    {
+        public static ErrorListService Instance = new ErrorListService();
+
+        public static void ProcessSarifLog(ResultLog resultLog)
+        {
+            foreach (RunLog runLog in resultLog.RunLogs)
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
```

Comment:
It is mostly a matter of taste. I agree that you can easily write cryptic code with LINQ, but in its simpler form I think it clarifies rather than obscures. IMO
```
IRuleDescriptor ruleDescriptor = runLog.RuleInfo.FirstOrDefault(rd => rd.Id == ruleId);
```
... being more compact that a loop, and being idiomatic LINQ, conveys information more quickly than having to read the loop to see what it does.

But I'm not advocating you change your style, or that we should impose these idioms as a style requirement -- really, just asking if you'd considered it.